library/std/src/io/util.rs

```diff
@@ -8,24 +8,41 @@ use crate::io::{
     self, BufRead, IoSlice, IoSliceMut, Read, ReadBuf, Seek, SeekFrom, SizeHint, Write,
 };
 
-/// A reader which is always at EOF.
+/// A writer which will move data into the void, and a reader which
+/// is always at EOF.
```

Comment:
```suggestion
/// `Empty` ignores any data written via [`Write`], and will always be empty
/// (returning zero bytes) when read via [`Read`].
```

This wording I think is a little confusing -- maybe we can directly address the trait impls instead?